File: dotnet/src/Zipwire.ProofPack/ProofPack/JwsSerializerOptions.cs
using System.Text.Json;
using Evoq.Blockchain.Merkle;

namespace Zipwire.ProofPack;

/// <summary>
/// Provides consistent JsonSerializerOptions for JWS operations.
/// </summary>
public static class JwsSerializerOptions
{
    /// <summary>
    /// Gets the default JsonSerializerOptions for JWS operations.
    /// This includes the MerkleTreeJsonConverter to ensure proper serialization of MerkleTree objects.
    /// </summary>
    /// <returns>The JsonSerializerOptions configured for JWS operations.</returns>
    public static JsonSerializerOptions GetDefault()
    {
        return new JsonSerializerOptions
        {
            WriteIndented = false,
            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
            DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingNull,
            Converters = { new MerkleTreeJsonConverter() }
        };
    }
}

File: javascript/packages/base/src/JwsSerializerOptions.js
/**
 * Provides consistent JSON serialization options for JWS operations.
 * 
 * This utility ensures that JWS serialization uses the correct options,
 * particularly for compact JSON output and camelCase property naming.
 */
export const JwsSerializerOptions = {
    /**
     * Gets the default JSON serialization options for JWS operations.
     * 
     * @returns {object} JSON serialization options configured for JWS operations
     */
    getDefault() {
        return {
            writeIndented: false,
            propertyNamingPolicy: 'camelCase',
            defaultIgnoreCondition: 'whenWritingNull'
        };
    }
}; 